lib/tasks/one_off.rake

```diff
@@ -285,17 +287,30 @@ namespace :one_off do
         affected_versions.destroy_all if commit
         puts "Trying manual sync for #{project.platform}/#{project.name}."
         project.try(:manual_sync) if commit
-
-        # Use versions count manually because it might be unsafe to trust
-        # cached project.versions_count value immediately after doing deletions
-        if project.versions.count == 0
-          puts "No versions remaining for #{project.platform}/#{project.name}: Destroying project."
-          project.destroy! if commit
-        end
       end
 
       processed_count += [batch_size, batch.size].min
       puts "Processed #{processed_count} projects."
     end
   end
+
+  # This should be run after the manual_syncs in delete_ignored_maven_versions_and_resync_packages are done
+  desc "Delete Maven packages without versions"
+  task :delete_maven_packages_without_versions, %i[commit] => :environment do |_t, args|
+    commit = args.commit.present? && args.commit == "yes"
+
+    puts "DRY RUN" unless commit
+
+    batch_size = 10000
+    processed_count = 0
+
+    affected_projects = Project.where(platform: "Maven").without_versions
```

Comment:
This can technically catch Maven projects that weren't affected by the `delete_ignored_maven_versions_and_resync_packages` task, but happen to have no versions anyway, and delete them. I am assuming this is likely fine, but if it isn't then we might need to be smarter about that.